core/templates/pages/exploration-editor-page/history-tab/services/compare-versions.service.spec.ts

```diff
@@ -864,107 +846,133 @@ describe('Compare versions service', function() {
+              stateProperty: 'unchanged',
+              originalStateName: 'A'
+            },
+            2: {
+              newestStateName: 'B',
+              stateProperty: 'deleted',
+              originalStateName: 'B'
+            }
+          });
         });
-        $httpBackend.flush();
-        expect(nodeData).toEqual({
-          1: {
-            newestStateName: 'A',
-            stateProperty: 'unchanged',
-            originalStateName: 'A'
-          },
-          2: {
-            newestStateName: 'B',
-            stateProperty: 'deleted',
-            originalStateName: 'B'
-          }
-        });
-      }
-    );
-
-    it('should mark states correctly when there are 2 reversions', function() {
-      $httpBackend.expect('GET', '/explorehandler/init/0?v=5')
-        .respond(_getStatesData(testExplorationData2[4]));
-      $httpBackend.expect('GET', '/explorehandler/init/0?v=8')
-        .respond(_getStatesData(testExplorationData2[7]));
-      vts.init(testSnapshots2);
-      cvs.getDiffGraphData(5, 8).then(function(data) {
-        expect(data.nodes).toEqual({
-          1: {
-            newestStateName: 'A',
-            stateProperty: 'unchanged',
-            originalStateName: 'A'
-          },
-          2: {
-            newestStateName: 'C',
-            stateProperty: 'added',
-            originalStateName: 'B'
-          },
-          3: {
-            newestStateName: 'D',
-            stateProperty: 'added',
-            originalStateName: 'D'
-          }
+
+        let req = httpTestingController.expectOne('/explorehandler/init/0?v=4');
+        expect(req.request.method).toEqual('GET');
+        req.flush(_getStatesData(testExplorationData2[3]));
+
+        let req2 = httpTestingController.expectOne(
+          '/explorehandler/init/0?v=5');
+        expect(req2.request.method).toEqual('GET');
+        req2.flush(_getStatesData(testExplorationData2[4]));
+
+        flushMicrotasks();
+      }));
+
+    it(
+      'should mark states correctly when there are 2 reversions',
+      fakeAsync(() => {
+        vts.init(testSnapshots2);
+        cvs.getDiffGraphData(5, 8).then(function(data) {
+          expect(data.nodes).toEqual({
+            1: {
+              newestStateName: 'A',
+              stateProperty: 'unchanged',
+              originalStateName: 'A'
+            },
+            2: {
+              newestStateName: 'C',
+              stateProperty: 'added',
+              originalStateName: 'B'
+            },
+            3: {
+              newestStateName: 'D',
+              stateProperty: 'added',
+              originalStateName: 'D'
+            }
+          });
         });
-      });
-      $httpBackend.flush();
-    });
+
+        let req = httpTestingController.expectOne('/explorehandler/init/0?v=5');
+        expect(req.request.method).toEqual('GET');
+        req.flush(_getStatesData(testExplorationData2[4]));
+
+        let req2 = httpTestingController.expectOne(
```

Comment:
Please use const for the req objects in this file.

Reply:
done.